### PR TITLE
Updated the LineGraphWidget with new layout

### LINKED ISSUE #16068: Earnings/Army graph values are not synced with game time on nondefault gamespeeds.

### LINKED ISSUE #16597: Army and earnings graph overflow content area
With heavy support starting units:

armygr

earnings

Release-20190314 with heavy support starting units:

armygraph_0314

### PR DESCRIPTION
**Pull request by teinarss**
Closes #16597 
Closes #16068

image

### CODE REVIEW

**Review by abcdefg30 — COMMENTED** [review]
Can we change `m` to `min` while we are at it? `m` usually refers to Meter which is not what we want in this context, I think.

**Comment by teinarss** [comment]
You are right, https://en.wikipedia.org/wiki/Non-SI_units_mentioned_in_the_SI

**Comment by teinarss** [comment]
Updated

**Comment by pchote** [comment]
Can we please move the player list to the right, add axis labels ("Army Value" / "Game Minute", "Earnings" / "Game Minute"), and drop the then-redundant title row? This would make the graphs a lot more conventional.

**Comment by pchote** [comment]
See 63069bcd951e28a510bcc7bc52ddc495623e9511 for drawing rotated text. Use an angle of `(float)Math.Pi / 2`.

**Comment by teinarss** [comment]
Updated and fixed the updates to actual use game minute

File: OpenRA.Mods.Common/Widgets/LineGraphWidget.cs [hunk 1]
```diff
@@ -131,41 +151,53 @@ public override void Draw()
...
+
+				var textWidth = labelFont.Measure(text);
+
+				var yLocation = y + (textWidth.Y + labelFont.TopOffset) / 2;
+
+				labelFont.DrawTextWithShadow(text, graphOrigin + new float2(-(textWidth.X + 3), -yLocation), Color.White, BackgroundColorDark, BackgroundColorLight, 1);
 			}
 
-			bold.DrawTextWithShadow(GetYAxisLabel(), origin + new float2(width + 40, -(height / 2)), Color.White, BackgroundColorDark, BackgroundColorLight, 1);
+			// bottom line
```
Comment (pchote) [inline OpenRA.Mods.Common/Widgets/LineGraphWidget.cs]: `// Bottom line`

File: OpenRA.Mods.Common/Widgets/LineGraphWidget.cs [hunk 2]
```diff
@@ -131,41 +151,53 @@ public override void Draw()
...
+			cr.DrawLine(graphOrigin, graphOrigin + new float2(width, 0), 1, Color.White);
 
-			cr.DrawLine(origin, origin + new float2(width, 0), 1, Color.White);
-			cr.DrawLine(origin, origin + new float2(0, -height), 1, Color.White);
-			cr.DrawLine(origin + new float2(width, 0), origin + new float2(width, -height), 1, Color.White);
-			cr.DrawLine(origin + new float2(0, -height), origin + new float2(width, -height), 1, Color.White);
+			// left line
```
Comment (pchote) [inline OpenRA.Mods.Common/Widgets/LineGraphWidget.cs]: `// Left line`

**Review by pchote — DISMISSED** [review]
Huge improvement over the current graphs. Great work, and thanks for putting up with my scope creeping. Just a couple of minor code style suggestions then :+1:

**Comment by teinarss** [comment]
Updated

**Review by pchote — APPROVED** [review]

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/b7f2ecf4b4b04e5fd13fa3c79c5254f8db0c5182...78d74618eb9b81a396eda367caba510c452d8763

File: OpenRA.Mods.Common/Traits/Player/PlayerStatistics.cs [hunk 3]
```diff
@@ -84,7 +86,9 @@ void UpdateArmyThisMinute()
 
 		void ITick.Tick(Actor self)
 		{
-			if (self.World.WorldTick % 1500 == 1)
+			var timestep = self.World.IsReplay ? replayTimestep : self.World.Timestep;
+
+			if (timestep * self.World.WorldTick % 60000 == 0)
```
Comment (GraionDilach) [inline OpenRA.Mods.Common/Traits/Player/PlayerStatistics.cs]: This breaks if 60000 can not be divided properly with the timestep.  Example case: fast gamespeed in RA.

**Pull request merged**